Handle vera remote controls, and devices without a last_scene_time

alanfischer opened the pull request:
## Description:
Fixes issues with vera remote control devices so they properly report their button status as sensor

alanfischer committed:
Handle vera remote controls, as well as handle devices without a last_scene_time

balloob reviewed (changes requested):
Remotes should not be made sensors, instead they should fire events when buttons pressed.

dshokouhi committed:
Update neato to support new StateVacuumDevice (#16035)

* Update neato to support new vacuum states

* Remove changes submitted in error

* Lint

* Review comments and fix resume cleaning

* Lint

klada committed:
vacuum/xiaomi_miio: Expose "sensor_dirty_left" attribute (#16003)

balloob committed:
Add forgiving add column (#16057)

* Add forgiving add column

* Lint

balloob committed:
Add forgiving add column (#16057)

* Add forgiving add column

* Lint

huangyupeng committed:
Tuya fix login problem and add login platform param (#16058)

* add a platform param to distinguish different app's account.

* fix requirements

huangyupeng committed:
Tuya fix login problem and add login platform param (#16058)

* add a platform param to distinguish different app's account.

* fix requirements

balloob committed:
Add notify platforms to loaded components (#16063)

balloob committed:
Add notify platforms to loaded components (#16063)

balloob committed:
Column syntax fix + Add a file if migration in progress (#16061)

* Add a file if migration in progress

* Warning

* Convert message for migration to warning

balloob committed:
Column syntax fix + Add a file if migration in progress (#16061)

* Add a file if migration in progress

* Warning

* Convert message for migration to warning

balloob committed:
Bumped version to 0.76.1

balloob committed:
Merge pull request #16064 from home-assistant/rc

0.76.1

scop committed:
Grammar and spelling fixes (#16065)

balloob committed:
Update frontend to 20180820.0

balloob committed:
Update frontend to 20180820.0

balloob committed:
Use aiohttp web.AppRunner (#16020)

* Use aiohttp web.AppRunner

* Stop site

appzer committed:
Update pushsafer.py (#16060)

balloob committed:
Alexa: context + log events (#16023)

leppa committed:
openuv: Add Current UV Level to list of conditions (#16042)

Calculated from UV index, based on table from
https://www.openuv.io/kb/uv-index-levels-colors

TimBailey-pnk committed:
TpLink Device Tracker Error (#15918)

* Fix 'Error setting up platform tplink' error when a scanner fails even if another scanner would succeed

* Try to fix tox errors

* Adjust code based on PR comments

balloob committed:
Deprecated stuff (#16019)

* Use async with for locks

* Fix regex in template test

* Close session correctly

* Use correct current_task method

* push camera cleanup

* Lint

* Revert current_task

* Update websocket_api.py

* Mock executor_job betteR

* Fix async_create_task mock

balloob committed:
Add recent context (#15989)

* Add recent context

* Add async_set_context to components not using new services

OverloadUT committed:
Add ecovacs component (#15520)

* Ecovacs Deebot vacuums

* All core features implemented

Getting fan speed and locating the vac are still unsupported until sucks adds support

* Move init queries to the added_to_hass method

* Adding support for subscribing to events from the sucks library

This support does not exist in sucks yet; this commit serves as a sort of TDD approach of what such support COULD look like.

* Add OverloadUT as ecovacs code owner

* Full support for Ecovacs vacuums (Deebot)

* Add requirements

* Linting fixes

* Make API Device ID random on each boot

* Fix unique ID

Never worked before, as it should have been looking for a key, not an attribute

* Fix random string generation to work in Python 3.5 (thanks, Travis!)

* Add new files to .coveragerc

* Code review changes

(Will require a sucks version bump in a coming commit; waiting for it to release)

* Bump sucks to 0.9.1 now that it has released

* Update requirements_all.txt as well

* Bump sucks version to fix lifespan value errors

* Revert to sucks 0.9.1 and include a fix for a bug in that release

Sucks is being slow to release currently, so doing this so we can get a version out the door.

* Switch state_attributes to device_state_attributes

foxel committed:
Bumped NDMS2 client library to 0.0.4 to get compatible with python 3.5 (#16077)

Danielhiversen committed:
fritzdect change to current_power_w (#16079)

danielperna84 committed:
Update pyhomematic to 0.1.47 (#16083)

awarecan committed:
Get user after login flow finished (#16047)

* Get user after login flow finished

* Add optional parameter 'type' to /auth/login_flow

* Update __init__.py

scop committed:
Upgrade pytest to 3.7.2 (#16091)

balloob committed:
Forgiving add index in migration (#16092)

balloob committed:
Forgiving add index in migration (#16092)

balloob committed:
Bumped version to 0.76.2

balloob committed:
Merge pull request #16094 from home-assistant/rc

0.76.2

fliphess committed:
Add verify ssl to generic camera (#15949)

* Add verify_ssl option to generic camera

* Remove flake8 errors

* Add test for ssl verification on and off

* Fix lint errors

zhelev committed:
Upgrade afsapi to 0.0.4, prevents aiohttp session close message, Fixes #13099 (#16098)

balloob committed:
Remove homeassistant.remote (#16099)

* Remove homeassistant.remote

* Use direct import for API

* Fix docstring

balloob committed:
Use new session when fetching remote urls (#16093)

awarecan committed:
Allow finish_flow callback to change data entry result type (#16100)

* Allow finish_flow callback to change data entry result type

* Add unit test

balloob committed:
Add support for revoking refresh tokens (#16095)

* Add support for revoking refresh tokens

* Lint

* Split revoke logic in own method

* Simplify

* Update docs

awarecan committed:
Refactoring login flow (#16104)

* Abstract LoginFlow

* Lint and typings

fabaff committed:
Minor updates (#16106)

syssi committed:
Bump python-miio version (#16110)

klada committed:
homematic: Add homematic.put_paramset service (#16024)

Service to call putParamset method of XML-RPC API

fabaff committed:
Upgrade shodan to 1.9.1 (#16113)

fabaff committed:
Upgrade sendgrid to 5.6.0 (#16111)

teharris1 committed:
Merge insteon_plm and insteon_local to insteon component (#16102)

* Implement X10

* Add X10 after add_device_callback

* Ref device by id not hex and add x10OnOffSwitch name

* X10 services and add sensor device

* Correctly reference X10_HOUSECODE_SCHEMA

* Log adding of X10 devices

* Add X10 All Units Off, All Lights On and All Lights Off devices

* Correct ref to X10 states vs devices

* Add X10 All Units Off, All Lights On and All Lights Off devices

* Correct X10 config

* Debug x10 device additions

* Config x10 from bool to housecode char

* Pass PLM to X10 device create

* Remove PLM to call to add_x10_device

* Unconfuse x10 config and method names

* Correct spelling of x10_all_lights_off_housecode

* Bump insteonplm to 0.10.0 to support X10

* Add host to config options

* Add username and password to config for hub connectivity

* Add username and password to config for hub

* Convert port to int if host is defined

* Add KeypadLinc

* Update config schema to require either port or host

* Solidify Hub and PLM configuration to ensure proper settings

* Update hub schema

* Bump insteonplm version

* Fix pylint and flake issues

* Bump insteonplm to 0.12.1

* Merge insteon_plm and insteon_local to insteon

* Rename insteon_plm to insteon

* Bump insteonplm to 0.12.2

* Flake8 cleanup

* Update .coveragerc for insteon_plm, insteon_local and insteon changes

* Add persistent notification

* Fix reference to insteon_plm

* Fix indentation

* Shorten message and fix grammer

* Add comment to remove in release 0.90

* Hound fix

jeradM committed:
Remove unit_of_measurement from climate entities (#16012)

* Remove unit_of_measurement from climate base class.

* Updated google_assistant component and tests to use core temp units.

* Fixes

* Convert Alexa component to use core temp units for climate entities.

* Fix tests.

* Converted prometheus component.

* Remove unit_of_measurement from homekit thermostat tests.

* Small fix.

awarecan committed:
Add multi-factor authentication modules (#15489)

* Get user after login flow finished

* Add multi factor authentication support

* Typings

Kane610 committed:
Device Registry (#15980)

* First draft

* Generate device id

* No obscure registry

* Dont store config_entry_id in device

* Storage

* Small mistake on rebase

* Do storage more like entity registry

* Improve device identification

* Add tests

* Remove deconz device support from PR

* Fix hound comments, voff!

* Fix comments and clean up

* Fix proper indentation

* Fix pydoc issues

* Fix mochad component to not use self.device

* Fix mochad light platform to not use self.device

* Fix TankUtilitySensor to not use self.device

* Fix Soundtouch to not use self.device

* Fix Plex to not use self.device

* Fix Emby to not use self.device

* Fix Heatmiser to not use self.device

* Fix Wemo lights to not use self.device

* Fix Lifx to not use self.device

* Fix Radiotherm to not use self.device

* Fix Juicenet to not use self.device

* Fix Qwikswitch to not use self.device

* Fix Xiaomi miio to not use self.device

* Fix Nest to not use self.device

* Fix Tellduslive to not use self.device

* Fix Knx to not use self.device

* Clean up a small mistake in soundtouch

* Fix comment from Ballob

* Fix bad indentation

* Fix indentatin

* Lint

* Remove unused variable

* Lint

balloob committed:
Check correctly if package is loadable (#16121)

fabaff committed:
Upgrade numpy to 1.15.1 (#16126)

fabaff committed:
Upgrade youtube_dl to 2018.08.22 (#16125)

syssi committed:
Fix the protocol v2 data_key of several aqara devices (#16112)

* Fix the protocol v2 data_key of several aqara devices

* Incorporate review

eavanvalkenburg committed:
Upgrade brunt package (#16130)

* Bumped package version to fix a bug with Python before 3.6

* update of package version in requirements_all

jcconnell committed:
Add support for NOAA tide information (new PR) (#15947)

* Adding noaa-tides changes to new branch.

* Fix typo in .coverageac

* Incorporate @MartinHjelmare and @fabaff changes.

* Disable pylint error and add error message for unavailable station.

* Two spaces before inline comments

* Increment py_noaa version to 0.3.0

* Updated requirements.py

* Minor changes

dgomes committed:
Adds support for routers implementing IGDv2 (#16108)

* Adds support for IGDv2

* bump pyupnp_async

* bump pyupnp_async

* better debug

* fix test

villanyibalint committed:
Add support for JS modules in custom panels (#16096)

* Added backend support for JavaScript modules in custom panels.

* Fixed test_panel_custom.py

* Delete core.entity_registry

* Update panel_custom.py

* Corrected panel_custom.py with module_url.

* Rebase

* Missed elif

* Add vol.Exclusive module_url

* Correct vol.Exclusive usage

* Test for js module

* Corrected line continuation indentation

* Added webcomponent path to exclusive group

* Corrected line length

* Line break

* Test for conflicting url options

* Self -> hass fix

* Fix self -> hass again

* Use assert_setup_component

* Setup missing

* Correct test

* Fix again

* Fix

* Mising async

* Fix

* test real

* Test real

* Final

* check

* Final check

* safety

* Final commit and check

* Removed unused dependencies

* Test for multiple url options in config

balloob committed:
Prevent legacy api password with empty password (#16127)

* Prevent legacy api password with empty password

* Typing

balloob committed:
Enable auth by default 🙈 (#16107)

* Enable auth by default

* Only default legacy_api_password if api_password set

* Tweak bool check

* typing

balloob committed:
Remove commented out API password from default config (#16147)

scop committed:
Spelling fixes (#16150)

scop committed:
Update pydocstyle to 2.1.1 and flake8-docstrings to 1.3.0 (#14557)

* Update pydocstyle to 2.1.1 and flake8-docstrings to 1.3.0

* Pydocstyle D401 fixes

hobbypunk90 committed:
Hangouts (#16049)

* add a component for hangouts

* add a notify component for hangouts

* add an extra message as title

* add support to listen to all conversations hangouts has

* move hangouts to package and add parameter documentation

* update .coveragerc and requirements_all.txt

* makes linter happy again

* bugfix

* add conversations parameter to command words

* Move the resolution of conversation names to conversations in own a function

* typo

* rename group of exclusion form 'id' to 'id or name'

* refactoring and use config_flow

* makes linter happy again

* remove unused imports

* fix not working regex commands

* fix translations

* cleanup

* remove step_init

* remove logging entry

* clean up events

* move constant

* remove unsed import

* add new files to .converagerc

* isort imports

* add hangouts_utils to ignored packages

* upadte doc and format

* fix I/O not in executor jon

* rename SERVICE_UPDATE_USERS_AND_CONVERSATIONS to SERVICE_UPDATE

* move EVENT_HANGOUTS_{CONNECTED,DISCONNECTED} to dispatcher

* add config flow tests

* Update tox.ini

balloob committed:
Bump frontend to 20180824.0

balloob committed:
Update translations

emlove committed:
Hangouts localization typo fix (#16174)

Kane610 committed:
deCONZ - Allow sub second light transitions (#16170)

Solves https://github.com/home-assistant/home-assistant/issues/16075

balloob committed:
add_devices -> add_entities (#16171)

* add_devices -> add_entities

* Lint

* PyLint

* Revert external method in scsgate

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Bumped version to 0.77.0b0

awarecan committed:
Add multi-factor auth module setup flow (#16141)

* Add mfa setup flow

* Lint

* Address code review comment

* Fix unit test

* Add assertion for WS response ordering

* Missed a return

* Remove setup_schema from MFA base class

* Move auth.util.validate_current_user -> webscoket_api.ws_require_user

awarecan committed:
Add multi-factor auth module setup flow (#16141)

* Add mfa setup flow

* Lint

* Address code review comment

* Fix unit test

* Add assertion for WS response ordering

* Missed a return

* Remove setup_schema from MFA base class

* Move auth.util.validate_current_user -> webscoket_api.ws_require_user

Kane610 committed:
deCONZ - Support device registry (#16115)

Add support for device registry in deCONZ component

Kane610 committed:
deCONZ - Support device registry (#16115)

Add support for device registry in deCONZ component

heythisisnate committed:
fix error message for cv.matches_regex (#16175)

heythisisnate committed:
fix error message for cv.matches_regex (#16175)

heythisisnate committed:
Decouple Konnected entity setup from discovery (#16146)

* decouple entity setup from discovery

* validate that device_id is a full MAC address

heythisisnate committed:
Decouple Konnected entity setup from discovery (#16146)

* decouple entity setup from discovery

* validate that device_id is a full MAC address

fabaff committed:
Add 'moon_phase' to Dark Sky sensor (#16179)

awarecan committed:
Tweak log level for bearer token warning (#16182)

awarecan committed:
Tweak log level for bearer token warning (#16182)

Kane610 committed:
Device registry store config entry (#16152)

* Allow device registry to optionally store config entries

* Connections and identifiers are now sets with tupels

* Make config entries mandatory

* Fix duplicate keys in test

* Rename device to device_info

* Entity platform should only create device entries if config_entry_id exists

* Fix Soundtouch tests

* Revert soundtouch to use self.device

* Fix baloobs comments

* Correct type in test

Kane610 committed:
Device registry store config entry (#16152)

* Allow device registry to optionally store config entries

* Connections and identifiers are now sets with tupels

* Make config entries mandatory

* Fix duplicate keys in test

* Rename device to device_info

* Entity platform should only create device entries if config_entry_id exists

* Fix Soundtouch tests

* Revert soundtouch to use self.device

* Fix baloobs comments

* Correct type in test

balloob committed:
Fix hangouts (#16180)

balloob committed:
Fix hangouts (#16180)

awarecan committed:
Default load trusted_network auth provider if configured trusted networks (#16184)

awarecan committed:
Default load trusted_network auth provider if configured trusted networks (#16184)

balloob committed:
Bump frontend to 20180825.0

balloob committed:
Bump frontend to 20180825.0

balloob committed:
Bumped version to 0.77.0b1

djm300 committed:
Zoneminder SSL fix (#16157)

* Update zoneminder.py

Added a verify_ssl parameter for zoneminder

* PEP8 fixup

* PEP8 indenting fix

* Fix lint issue

* Remove whitespace

thomasdelaet committed:
update python-velbus library version (#16194)

smurfix committed:
MQTT: Log transmitted as well as received messages (#16195)

klada committed:
homematic: Make device avilable again when UNREACH becomes False (#16202)

klada committed:
homematic: Make device avilable again when UNREACH becomes False (#16202)

fabaff committed:
Upgrade sqlalchemy to 1.2.11 (#16192)

fucm committed:
Add battery warning, rssi level and check for availability (#16193)

Kane610 committed:
Revert changes to platforms using self.device (#16209)

* Revert tank_utility

* Fix Soundtouch

* Fix Plex

* Fix Emby

* Fix Radiotherm

* Fix Juicenet

* Fix Qwikswitch

* Fix Xiaomi miio

* Fix Nest

* Fix Tellduslive

* Fix KNX

Kane610 committed:
Revert changes to platforms using self.device (#16209)

* Revert tank_utility

* Fix Soundtouch

* Fix Plex

* Fix Emby

* Fix Radiotherm

* Fix Juicenet

* Fix Qwikswitch

* Fix Xiaomi miio

* Fix Nest

* Fix Tellduslive

* Fix KNX

grea09 committed:
Inconsistent entity_id when multiple sensors (#16205)

* Inconsistent entity_id when multiple sensors

I am submitting a change to fix a [bug](https://github.com/home-assistant/home-assistant/issues/16204) for when there are several sensors for the same hostname. For example I want to track my IPv4 and IPv6 address. It creates two entities that regularly switch ids based on the order they get initialized.

To fix this I comform to the way other componnents have addressed the issue by adding an optional `name` attribute.

* Line too long

* Removing trailing whitespace

hobbypunk90 committed:
remove hangouts.users state, simplifies hangouts.conversations (#16191)

hobbypunk90 committed:
remove hangouts.users state, simplifies hangouts.conversations (#16191)

PhracturedBlue committed:
Handle exception from pillow (#16190)

PhracturedBlue committed:
Handle exception from pillow (#16190)

balloob committed:
Update frontend to 20180826.0

balloob committed:
Update frontend to 20180826.0

Swamp-Ig committed:
Update aiohttp to version 3.4.0. (#16198)

Swamp-Ig committed:
Update aiohttp to version 3.4.0. (#16198)

awarecan committed:
Add Time-based Onetime Password Multi-factor Authentication Module (#16129)

* Add Time-based Onetime Password Multi-factor Auth

Add TOTP setup flow, generate QR code

* Resolve rebase issue

* Use svg instead png for QR code

* Lint and typing

* Fix translation

* Load totp auth module by default

* use <svg> tag instead markdown image

* Update strings

* Cleanup

awarecan committed:
Add Time-based Onetime Password Multi-factor Authentication Module (#16129)

* Add Time-based Onetime Password Multi-factor Auth

Add TOTP setup flow, generate QR code

* Resolve rebase issue

* Use svg instead png for QR code

* Lint and typing

* Fix translation

* Load totp auth module by default

* use <svg> tag instead markdown image

* Update strings

* Cleanup

Eriner committed:
Replace pbkdf2 with bcrypt (#16071)

* Replace pbkdf2 with bcrypt

bcrypt isn't inherently better than pbkdf2, but everything "just works"
out of the box.

  * the hash verification routine now only computes one hash per call
  * a per-user salt is built into the hash as opposed to the current
  global salt
  * bcrypt.checkpw() is immune to timing attacks regardless of input
  * hash strength is a function of real time benchmarks and a
  "difficulty" level, meaning we won't have to ever update the iteration
  count

* WIP: add hash upgrade mechanism

* WIP: clarify decode issue

* remove stale testing code

* Fix test

* Ensure incorrect legacy passwords fail

* Add better invalid legacy password test

* Lint

* Run tests in async scope

Eriner committed:
Replace pbkdf2 with bcrypt (#16071)

* Replace pbkdf2 with bcrypt

bcrypt isn't inherently better than pbkdf2, but everything "just works"
out of the box.

  * the hash verification routine now only computes one hash per call
  * a per-user salt is built into the hash as opposed to the current
  global salt
  * bcrypt.checkpw() is immune to timing attacks regardless of input
  * hash strength is a function of real time benchmarks and a
  "difficulty" level, meaning we won't have to ever update the iteration
  count

* WIP: add hash upgrade mechanism

* WIP: clarify decode issue

* remove stale testing code

* Fix test

* Ensure incorrect legacy passwords fail

* Add better invalid legacy password test

* Lint

* Run tests in async scope

balloob committed:
Update translations

balloob committed:
Update translations

balloob committed:
Add new translations

balloob committed:
Add new translations

balloob committed:
Bumped version to 0.77.0b2

cereal2nd committed:
Add temperature sensors to the velbus component (#16203)

* Added support for velbus temperature sensors

* Bumped the required version

* updated requirements_all.txt

* Auto review comments fixed

* Updated after comments

* Updated after comments

* Fix travis

* Fix travis

endor-force committed:
Rewrite of Trafikverket weather - Multiple sensor types supported (#15935)

* Added precipitation type from API

Enables users to see type of precipitation.
Value returned from API is a string in swedish.

* Corrected tox verification errors

Correction of tox findings

* Missed in tox - fixed

* Hound witespace fix

* Updated comment to trigger travis rebuild

Travis tox failed due to problem with tox build process. 
Correcting in a comment to trigger retry in travis..

* Try to retrigger travis/tox successful rebuild

* Cleaning

* Cleaning more

* Trafikverket rebuilt for library

Extended pytrafikverket with weather sensor collction
Changed behaviour of sensor component to use pytrafikverket.
Added more sensors.

User need to change config to use new version.
 [] Documentation needs to be updated

* Cleaned up based on Martins input

Appreciate the feedback

kariudo committed:
Add device_tracker.bluetooth_update service (#15252)

* Add device_tracker.bluetooth_update service

Will immediately scan for Bluetooth devices outside of the interval timer. Allows for less frequent scanning, with scanning on demand via automation.

* remove excess whitespace per bot comments

* Refactored update_bluetooth to call new function update_bluetooth_once

* Change service name to bluetooth_tracker_update to reflect platform name

* Reformat for line length

* Linting fix, pydoc, first line should end with a period

* Fixed a method call, and removed some more unsused parameters

JulianKahnert committed:
Fix geizhals price parsing (#15990)

* fix geizhals price parsing

* Fix lint issue

* switch to the geizhals pypi package

* throttle updates

* update geizhals version

* initialize empty device

* minor changes to trigger another TravisCI test

* device => _device

* bump geizhals version

balloob committed:
Update translations

balloob committed:
Update translations

balloob committed:
Change auth warning (#16216)

balloob committed:
Change auth warning (#16216)

fabaff committed:
Add Volkszaehler sensor (#16188)

* Add Volkszaehler sensor

* Update icons

* Improve code

vrih committed:
openalpr flag `WITH_TEST` should be `WITH_TESTS` (#16218)

Removes warning from openalpr build and saves a few seconds from build
time as tests weren't being bypassed as intended

balloob committed:
Fix device telldus (#16224)

balloob committed:
Fix device telldus (#16224)

fabaff committed:
Upgrade alpha_vantage to 2.1.0 (#16217)

balloob committed:
Bump frontend to 20180827.0

balloob committed:
Bump frontend to 20180827.0

hobbypunk90 committed:
rewrite hangouts to use intents instead of commands (#16220)

* rewrite hangouts to use intents instead of commands

* small fixes

* remove configured_hangouts check and CONFIG_SCHEMA

* Lint

* add import from .config_flow

hobbypunk90 committed:
rewrite hangouts to use intents instead of commands (#16220)

* rewrite hangouts to use intents instead of commands

* small fixes

* remove configured_hangouts check and CONFIG_SCHEMA

* Lint

* add import from .config_flow

Kane610 committed:
Store devices as dict instead of list (#16229)

* Store devices as dict instead of list

* Use OrderedDict

balloob committed:
Update trusted networks flow (#16227)

* Update the trusted networks flow

* Fix tests

* Remove errors

balloob committed:
Update trusted networks flow (#16227)

* Update the trusted networks flow

* Fix tests

* Remove errors

balloob committed:
Bumped version to 0.77.0b3

balloob committed:
Warning missed a space (#16233)

balloob committed:
Warning missed a space (#16233)

balloob committed:
Package loadable: compare case insensitive (#16234)

balloob committed:
Package loadable: compare case insensitive (#16234)

awarecan committed:
Change log level to error when auth provider failed loading (#16235)

awarecan committed:
Change log level to error when auth provider failed loading (#16235)

balloob committed:
Avoid insecure pycryptodome (#16238)

balloob committed:
Avoid insecure pycryptodome (#16238)

balloob committed:
Improve package loadable (#16237)

* Add caching to package loadable

* Fix tests

* Improve package loadable

* Lint

* Typing

balloob committed:
Fix hangouts (#16232)

balloob committed:
Fix hangouts (#16232)

awarecan committed:
Blow up startup if init auth providers or modules failed (#16240)

* Blow up startup if init auth providers or modules failed

* Delete core.entity_registry

awarecan committed:
Blow up startup if init auth providers or modules failed (#16240)

* Blow up startup if init auth providers or modules failed

* Delete core.entity_registry

balloob committed:
Update translations

balloob committed:
Update translations

balloob committed:
Bumped version to 0.77.0b4

Kane610 committed:
def device shouldnt call it self but self._device (#16255)

Kane610 committed:
def device shouldnt call it self but self._device (#16255)

awarecan committed:
Tweak MFA login flow (#16254)

* Tweak MFA login flow

* Fix typo

awarecan committed:
Tweak MFA login flow (#16254)

* Tweak MFA login flow

* Fix typo

balloob committed:
Bump frontend to 20180829.0

balloob committed:
Bump frontend to 20180829.0

balloob committed:
Update translations

balloob committed:
Update translations

balloob committed:
Version bump to 0.77.0

balloob committed:
Merge pull request #16256 from home-assistant/rc

0.77

balloob committed:
Merge branch 'master' into dev

balloob committed:
Version bump to 0.78.0.dev0

Danielhiversen committed:
Switchmate (#15535)

* Switchmate

* switchmate

* swithcmate

* switchmate

* switchmate

* fix comments

* Update switchmate.py

* change error log

schmittx committed:
Add support for discrete states to MyQ cover (#16251)

* Add discrete states and update dependency

* Add translation dict

dgomes committed:
bump version (#16262)

cgtobi committed:
Replace Authorization by Authentication (#16259)

balloob committed:
Add device info to Chromecast (#16261)

balloob committed:
Add device info for sonos (#16263)

* Add device info for sonos

* Sets

balloob committed:
Add device info Nest (#16265)

* Add device info Nest

* Sets

balloob committed:
Add Hue device info (#16267)

* Add Hue device info

* Set with tuples

* Fix tests

awarecan committed:
Nest Thermostat has software version (#16275)

asmfreak committed:
Add support for Habitica (#15744)

* Added support for Habitica

Second refactoring
Moved all config to component.
Sensors are autodiscovered.
Signed-off-by: delphi <cpp.create@gmail.com>

* Apply requested changes

Signed-off-by: delphi <cpp.create@gmail.com>

* Made event fire async. Made `sensors` config implicit and opt-out-style.

Signed-off-by: delphi <cpp.create@gmail.com>

* Removed unneeded check and await.

Signed-off-by: delphi <cpp.create@gmail.com>

* Moved into separate component package and added service.yaml

Signed-off-by: delphi <cpp.create@gmail.com>

* Fix coveralls

Signed-off-by: delphi <cpp.create@gmail.com>

syssi committed:
Fix data_key override by parent class (#16278)

syssi committed:
Fix data_key override by parent class (#16278)

balloob committed:
Bump frontend to 20180829.1

balloob committed:
Bump frontend to 20180829.1

cnrd committed:
Fix error when vacuum is idling (#16282)

cnrd committed:
Fix error when vacuum is idling (#16282)

Kane610 committed:
Fix so that entities are properly unloaded with config entry (#16281)

balloob committed:
Bumped version to 0.77.1

balloob committed:
Merge pull request #16284 from home-assistant/rc

0.77.1

balloob committed:
Merge branch 'master' into dev

robbiet480 committed:
Fix spelling mistake in recorder migration [ci skip]

exxamalte committed:
Geo Location component (#15953)

* initial working version of a geo location component and georss platform

* ensure that custom attributes don't override built-in ones

* bugfixes and tests

* fixing tests because of introduction of new component using same fixture

* improving test cases

* removing potentially unavailable attribute from debug message output

* completing test suite

* cleaning up debug messages; sorting entries in group view by distance

* ability to define the desired state attribute and corresponding unit of measurement; sort devices in group by configured state; find centroid for map if event is defined by polygon; updated tests

* sort entries in group; code clean-ups

* fixing indentation

* added requirements of new component and platform

* fixed various lint issues

* fixed more lint issues

* introducing demo geo location platform; refactored geo location component and geo rss platform to fit

* removing geo rss events platform; added unit tests for geo location platform and demo platform

* reverting change in debug message for feedreader to avoid confusion with new geo location component

* updated requirements after removing georss platform

* removed unused imports

* fixing a lint issue and a test case

* simplifying component code; moving code into demo platform; fixing tests

* removed grouping from demo platform; small refactorings

* automating the entity id generation (the use of an entity namespace achieves the same thing)

* undoing changes made for the georss platform

* simplified test cases

* small tweaks to test case

* rounding all state attribute values

* fixing lint; removing distance from state attributes

* fixed test

* renamed add_devices to add_entities; tweaked test to gain more control over the timed update in the demo platform

* reusing utcnow variable instead of patched method

* fixed test by avoiding to make assumptions about order of list of entity ids

* adding test for the geo location event class

rytilahti committed:
Bump songpal dependency (#16297)

Fixes #14936

PhracturedBlue committed:
Use asterisk_mbox 0.5.0 client (#16296)

amelchio committed:
Fix LIFX effects (#16309)

amelchio committed:
Fix LIFX effects (#16309)

exxamalte committed:
avoid error in debug log mode and rss entry without title (#16316)

exxamalte committed:
avoid error in debug log mode and rss entry without title (#16316)

lamiskin committed:
Correct wemo static device discovery issue. (#16292)

A recent change caused an issue if a single static wemo device is offline and could not be reached, then the whole component would not initialize (and therefore all other wemo devices are not added).

lamiskin committed:
Correct wemo static device discovery issue. (#16292)

A recent change caused an issue if a single static wemo device is offline and could not be reached, then the whole component would not initialize (and therefore all other wemo devices are not added).

balloob committed:
Update frontend to 20180831.0

balloob committed:
Update frontend to 20180831.0

balloob committed:
Bumped version to 0.77.2

balloob committed:
Update translations

balloob committed:
Update translations

balloob committed:
Merge pull request #16324 from home-assistant/rc

0.77.2

thomaslian committed:
Upgrade Adafruit-DHT to 1.3.4 (#16327)

* Update dht.py

* Update requirements_all.txt

schmittx committed:
Add unique ID (#16323)

teharris1 committed:
Add Cover to the Insteon component (#16215)

* Create cover platform

* Create insteon cover platform

* Bump insteonplm to 0.13.0

* Change async_add_devices to async_add_entities

* Missing doc string

* Simplify open and set_position

* Flake8

* Bump insteonplm to 0.13.1

* Code review changes

* Flake8 updates

danielperna84 committed:
Fix missing humidity sensor (#16337)

Commit:
Support Sonos Beam HDMI input (#16340)

pnbruckner committed:
Make last_seen attribute a timezone aware datetime in UTC (#16348)

The last_seen attribute was a datetime in the local timezone but with
no tzinfo (i.e., a "naive" datetime.) When state changes occurred it
would be printed incorrectly in homeassistant.log because
homeassistant.util.dt.as_local assumes any datetime without tzinfo is
UTC. Also most, if not all, datetime attributes are timezone aware in
UTC. So use homeassistant.util.dt.as_utc (which assumes a naive
datetime is local) to convert last_seen to a timezone aware datetime
in UTC.

Danielhiversen committed:
Handle netatmo exception (#16344)

Joshi425 committed:
Add support for sound_mode for Yamaha rxv media_player (#16352)

bieniu committed:
Add unique_id to MQTT Light (#16303)

* Add unique_id

* Delete whitespaces

jesserizzo committed:
Update to EnvoyReader 0.2, support for more hardware (#16212)

* Add support for older Envoy models

* Stop requiring envoy model name in config

* Update to envoy_reader0.2

* Minor formatting fixes

* run script/gen_requirements_all.py

* Minor formatting fixes

* Change some strings to constants, use getattr to call function

todschmidt committed:
Snips: Added special slot values, session_id and slotname_raw (#16185)

* Added special slot values, site_id, session_id, and slotname_raw

* Update snips.py

awarecan committed:
Upgrade hbmqtt to 0.9.4 (#16356)

* Upgrade to hbmqtt 0.9.4

* Lint

* Typo

fabaff committed:
Upgrade Sphinx to 1.7.7 (#16359)

fabaff committed:
Upgrade qnapstats to 0.2.7 (#16360)

fabaff committed:
Upgrade mutagen to 1.41.1 (#16361)

fabaff committed:
Upgrade shodan to 1.10.0 (#16363)

fabaff committed:
Upgrade to youtube_dl to 2018.09.01 (#16365)

MarcSN311 committed:
Added 'nomapnt', 'outcurnt', 'loadapnt' fields (#16176)

* Added 'nomapnt', 'outcurnt', 'loadapnt' fields

Also added Ampere and Volt-Ampere to INFERRED_UNITS

* Fix lint issue

aronsky committed:
Generic Thermostat: add support for climate.turn_on/climate.turn_off (#16080)

* Added async_turn_on and async_turn_off implementations.

* Added turning on/off tests to generic thermostat

* style

* style

* style

fucm committed:
Add support for Tahoma Lighting Receiver on/off io (#15925)

* Add support for Tahoma light switch

* Clean up attributes and add available method

* Remove else statement

htotoo committed:
Update discord.py (#16248)

* Update discord.py

* Update discord.py

Fixed ATTR_IMAGES checking, line length, and ATTR_DATA imported. Also fixed missing spaces.

* Update discord.py

Fix E302...

fabaff committed:
Upgrade python-telegram-bot to 11.0.0 (#16373)

fabaff committed:
Upgrade attrs to 18.2.0 (#16372)

awarecan committed:
Update server.py (#16375)

Danielhiversen committed:
add_entities for switchmate (#16368)

balloob committed:
Update frontend to 20180903.0

balloob committed:
Update frontend to 20180903.0

balloob committed:
Update translations

balloob committed:
Update translations

balloob committed:
Bumped version to 0.77.3

balloob committed:
Merge pull request #16390 from home-assistant/rc

0.77.3

balloob committed:
Merge branch 'master' into dev

pszafer committed:
Support for playing radio preset by Onkyo media_player (#16258)

* Added support to play radio preset by play media

* Switch radio station by preset for Onkyo

* added SUPPORT_PLAY_MEDIA

Danielhiversen committed:
Bug fix for Tibber (#16397)

grantclem commented:
Hi Alan, any progress?

rcloran committed:
zha: Bump to zigpy 0.2.0/bellows 0.7.0 (#16404)

ReneNulschDE committed:
Fix SystemMonitor IP address sensor (#16394)

9R committed:
add ExpressBus icon key to sensor.mvg (#16387)

Kane610 committed:
Device and entity registry remove config entry on unload (#16247)

* Test

* Ability to remove device

* Don't remove devices, instead remove config entry from device and entity registries

* Remove print

* Remove is not the same as unload

* Add tests

* Fix hound comment

bachya committed:
Add config flow for OpenUV (#16159)

* OpenUV config flow in place

* Test folder in place

* Owner-requested comments

* Tests

* More tests

* Owner-requested changes (part 1 of 2)

* Updated requirements

* Owner-requested changes (2 of 2)

* Removed unnecessary import

* Bumping Travis

* Updated requirements

* More requirements

* Updated tests

* Owner-requested changes

* Hound

* Updated docstring

Kane610 committed:
Service to load new deCONZ devices without restart (#16308)

* New service to load new devices from deCONZ without restarting HASS

* Do not use len to check if list is empty

* Add support for scenes to be updated as well

* Rework refresh devices method

* Fix test

rohankapoorcom committed:
Delegate mqtt topic match validation to the paho mqtt client (#16403)

* Delegate mqtt match topics to the paho mqtt client

* Fixing linting error with importing MQTTMatcher

Danielhiversen committed:
Clean up dlink and some bug fix (#16346)

* Update dlink.py

* style

* style

nudded committed:
Implement correct state for RFlink cover (#16304)

* implement correct state for rflink cover

* Fix linting error

* invert logic as local testing pointed out it should be reversed

* add period at the end to satisfy the linter

PaulAnnekov committed:
Fix Mi Flora median calculation (#16085)

* fixed median was based on 1.5 minute interval, not 1 hour

* ignore median and set state from first value, when previous state was None

* update before add, removed unused 'retries' and 'ble_timeout', check if platform ready

* added missing blank line

* fixed too long line

* using modern python 3.5 features, changed comment to be less verbose

* continuation line fix

* removed DEFAULT_SCAN_INTERVAL in favor of existing SCAN_INTERVAL

balloob committed:
Add context to scripts and automations (#16415)

* Add context to script helper

* Update script component

* Add context to automations

* Lint

awarecan committed:
Fix openuv.config_flow unit test (#16419)

alanfischer committed:
Added firing a vera_remote event when a remote or scene_controller is used

alanfischer committed:
Merge

homeassistant commented:
Hi @villanyibalint, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#15802).

Once you do that we will be able to review and accept this pull request.

Thanks!
